### PR TITLE
[SETUPAPI] CM_Request_Device_Eject_ExW: Allow ulNameLength to be zero when pszVetoName is not NULL

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose
Follow-up of PR [#5943](https://github.com/reactos/reactos/pull/5943)

Before:
ReactOS_Before

After:
ReactOS_After

### CODE REVIEW

File: dll/win32/setupapi/cfgmgr.c [hunk 1]
```diff
@@ -7548,7 +7548,15 @@ CM_Request_Device_Eject_ExW(
...
     if (pszVetoName == NULL && ulNameLength != 0)
+    {
         return CR_INVALID_POINTER;
+    }
+    else if (pszVetoName != NULL && ulNameLength == 0)
+    {
+        /* Windows 2003 SP2 ignores pszVetoName when ulNameLength is zero
+         * and behaves like when pszVetoName is NULL */
+        pszVetoName = NULL;
+    }
```
Comment (HBelusca) [inline dll/win32/setupapi/cfgmgr.c:7559]: Could this whole code be simplified as:
```suggestion
    /* Windows 2003 SP2 ignores pszVetoName when ulNameLength
     * is zero and behaves like when pszVetoName is NULL */
    if (ulNameLength == 0)
        pszVetoName = NULL;

    if (!pszVetoName && ulNameLength != 0)
        return CR_INVALID_POINTER;
```

**Review by HBelusca — APPROVED** [review]
I wonder whether this is also something other windows versions do.

A minor suggestion, but otherwise LGTM

File: dll/win32/setupapi/cfgmgr.c (hunk 1 above)
Comment (TAN-Gaming) [reply dll/win32/setupapi/cfgmgr.c]: Simplified.

**Review by katahiromz — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
LGTM. You can specify in the commit message `Addendum to 7f346b1aa`.

**Pull request merged**